### ISSUE TITLE
Please add gradient boosting

### ISSUE THREAD

**Issue by AndreyGurevich** [ypeleg/HungaBunga#1]
Please, add gradient boosting libraries, like CatBoost, XGBoost, LightGBM.

**Comment by ypeleg** [comment]
Currently working on gradient sklearn's boosting. 

In general I plan to add an optional sub-module for doing the same thing with third-party libraries. 
Then I will extend this to many other libraries such as catboost, xgboost and lightgbm.

**Comment by Hemang999** [comment]
Any idea when it will be completed?

**Comment by amueller** [comment]
I recommend to add HistGradientBoosting which should be about as fast as LightGBM and faster than XGBoost [though doesn't support row or column subsampling right now, work in progress].
